spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,286 @@
+        self._progress_bar.setValue(current_value)
+
+
+class UpdateInstallerDialog(QDialog):
+    """Update installer dialog."""
+
+    # Signal to get the download progress
+    # int: Download progress
+    # int: Total download size
+    sig_download_progress = Signal(int, int)
+
+    # Signal to get the current status of the update installation
+    # str: Status string
+    sig_installation_status = Signal(str)
+
+    def __init__(self, parent):
+
+        self.cancelled = False
+        self.status = NO_STATUS
+        self.thread_install_update = None
+        super().__init__(parent)
+        self.setWindowFlags(Qt.Dialog | Qt.MSWindowsFixedSizeDialogHint)
+        self._parent = parent
+        self._installation_widget = UpdateInstallation(self)
+        self.latest_release_version = ""
+        # Layout
+        installer_layout = QVBoxLayout()
+        installer_layout.addWidget(self._installation_widget)
+        self.setLayout(installer_layout)
+
+        # Signals
+        self.sig_download_progress.connect(
+            self._installation_widget.update_installation_progress)
+        self.sig_installation_status.connect(
+            self._installation_widget.update_installation_status)
+        self.sig_installation_status.connect(
+            self.finished_installation)
+
+        self._installation_widget.ok_button.clicked.connect(
+            self.close_installer)
+        self._installation_widget.cancel_button.clicked.connect(
+            self.cancel_install)
+
+        # Show installation widget
+        self.setup()
+
+    def setup(self):
+        """Setup visibility of widgets."""
+        self._installation_widget.setVisible(True)
+        self.adjustSize()
+
+    def cancel_install(self):
+        """Cancel the installation in progress."""
+        reply = QMessageBox.critical(
+            self._parent, 'Spyder',
+            _('Do you really want to cancel the Spyder update installation?'),
+            QMessageBox.Yes, QMessageBox.No)
+        if reply == QMessageBox.Yes:
+            self.cancelled = True
+            self.cancel_thread_install_update()
+            self.setup()
+            self.accept()
+            return True
+        return False
+
+    def continue_install(self):
+        """
+        Continue the installation in progress by downloading the installer.
+        """
+        reply = QMessageBox(icon=QMessageBox.Question,
+                            text=_("Would you like to automatically download "
+                                   "and install the latest version of Spyder?"
+                                   "<br><br>"),
+                            parent=self._parent)
+        reply.setWindowTitle("Spyder")
+        reply.setAttribute(Qt.WA_ShowWithoutActivating)
+        reply.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+        reply.exec_()
+        if reply.result() == QMessageBox.Yes:
+            self.start_installation_update()
```

Comment:
This line is triggering an `TypeError` exception since the `latest_release_version` arg is not being passed:

```
Traceback (most recent call last):
  File "C:\Users\dalth\AppData\Local\Programs\Spyder\pkgs\spyder\plugins\application\widgets\status.py", line 111, in show_installation_dialog_or_menu
    self.installer.continue_install()
  File "C:\Users\dalth\AppData\Local\Programs\Spyder\pkgs\spyder\plugins\application\widgets\install.py", line 191, in continue_install
    self.start_installation_update()
TypeError: start_installation_update() missing 1 required positional argument: 'latest_release_version'
```

I think we should, from the container, pass the `latest_release_version` to the status widget/installer widget even if the user doesn't accept immediatly the first dialog to update (the one that informs that a new version is available)